### PR TITLE
OCLOMRS-480: Edit the reference of a Concept in a Dictionary

### PR DESCRIPTION
**Pull request by brucemakallan**
# JIRA TICKET NAME:
[Edit the reference of a Concept in a Dictionary](https://issues.openmrs.org/browse/OCLOMRS-480)

# Summary:
After Editing a Concept, edit its corresponding reference in the current Dictionary as well.

### CODE REVIEW

**Comment by brucemakallan** [comment]
This Pull Request is a Work in Progress

File: src/tests/dictionaryConcepts/actions/dictionaryConcept.test.js [hunk 1]
```diff
@@ -669,6 +669,38 @@ describe('Testing Edit concept actions ', () => {
     });
   });
 
+  it('should unretire a Mapping', () => {
```
Comment (dkayiwa) [inline src/tests/dictionaryConcepts/actions/dictionaryConcept.test.js]: How is this related to the ticket?

File: src/tests/dictionaryConcepts/actions/dictionaryConcept.test.js [hunk 2]
```diff
@@ -669,6 +669,38 @@ describe('Testing Edit concept actions ', () => {
...
+      { type: UPDATE_CONCEPT, payload: existingConcept },
+    ];
+    const store = mockStore(mockConceptStore);
+    const conceptUrl = '/orgs/EthiopiaNHDD/sources/HMIS-Indicators/concepts/C1.1.1.1/';
+    return store.dispatch(unretireMapping(conceptUrl)).then(() => {
+      expect(store.getActions()).toEqual(expectedActions);
+    });
+  });
+
+  it('should handle exceptions for unretire a Mapping', () => {
```
Comment (dkayiwa) [inline src/tests/dictionaryConcepts/actions/dictionaryConcept.test.js]: The same as above.

File: src/tests/dictionaryConcepts/actions/dictionaryConcept.test.js [hunk 3]
```diff
@@ -62,6 +62,7 @@ import {
   removeSelectedAnswer,
   unpopulatePrepopulatedAnswers,
   prepopulateAnswers,
+  unretireMapping,
```
Comment (dkayiwa) [inline src/tests/dictionaryConcepts/actions/dictionaryConcept.test.js]: How is this related to the ticket?

File: src/components/dictionaryConcepts/containers/EditConcept.jsx [hunk 4]
```diff
@@ -506,5 +553,7 @@ export default connect(
     addSelectedAnswers: addSelectedAnswersToState,
     removeAnswer: removeSelectedAnswer,
     createNewAnswerRow: addNewAnswerRow,
+    deleteConcept: removeDictionaryConcept,
```
Comment (dkayiwa) [inline src/components/dictionaryConcepts/containers/EditConcept.jsx]: How are these two new lines related to the ticket?

File: src/components/dictionaryConcepts/containers/EditConcept.jsx [hunk 5]
```diff
@@ -178,7 +221,10 @@ export class EditConcept extends Component {
 
     const regx = /^[a-zA-Z\d-_]+$/;
     if (regx.test(this.state.id) && this.state.datatype && this.state.concept_class) {
-      this.props.updateConcept(this.conceptUrl, this.state, this.props.history);
+      const unRetiredMappings = mappings.filter(m => m.retired === false);
```
Comment (dkayiwa) [inline src/components/dictionaryConcepts/containers/EditConcept.jsx]: How is the above related to the ticket?

File: src/tests/dictionaryConcepts/actions/dictionaryConcept.test.js (hunk 1 above)
Comment (brucemakallan) [reply src/tests/dictionaryConcepts/actions/dictionaryConcept.test.js]: It was affecting my coverage but I have now created a separate [PR](https://github.com/openmrs/openmrs-ocl-client/pull/382) for it.

File: src/tests/dictionaryConcepts/actions/dictionaryConcept.test.js (hunk 2 above)
Comment (brucemakallan) [reply src/tests/dictionaryConcepts/actions/dictionaryConcept.test.js]: It was affecting my coverage but I have now created a separate [PR](https://github.com/openmrs/openmrs-ocl-client/pull/382) for it.

File: src/tests/dictionaryConcepts/actions/dictionaryConcept.test.js (hunk 3 above)
Comment (brucemakallan) [reply src/tests/dictionaryConcepts/actions/dictionaryConcept.test.js]: It was affecting my coverage but I have now created a separate [PR](https://github.com/openmrs/openmrs-ocl-client/pull/382) for it.

File: src/components/dictionaryConcepts/containers/EditConcept.jsx (hunk 4 above)
Comment (brucemakallan) [reply src/components/dictionaryConcepts/containers/EditConcept.jsx]: A reference to a concept cannot be edited directly, therefore, it has to first be deleted from the Dictionary, then Recreated. This prop is being called within the method: `updateConceptReference()`

File: src/components/dictionaryConcepts/containers/EditConcept.jsx [hunk 6]
```diff
@@ -64,6 +65,9 @@ export class EditConcept extends Component {
     selectedAnswers: PropTypes.array.isRequired,
     removeAnswer: PropTypes.func.isRequired,
     createNewAnswerRow: PropTypes.func.isRequired,
+    deleteConcept: PropTypes.func.isRequired,
```
Comment (dkayiwa) [inline src/components/dictionaryConcepts/containers/EditConcept.jsx]: How is the above line related to this ticket?

File: src/components/dictionaryConcepts/containers/EditConcept.jsx (hunk 5 above)
Comment (brucemakallan) [reply src/components/dictionaryConcepts/containers/EditConcept.jsx]: When recreating a reference to a concept, the API defaults all Mappings of that Concept to `retired: false` therefore this line is making sure that the retired mappings are not set to true.

**Comment by brucemakallan** [comment]
Hello @dkayiwa. The coverage drop is due to changes that are not directly related to this PR. However, this will be rectified after https://github.com/openmrs/openmrs-ocl-client/pull/382 is merged.

File: src/components/dictionaryConcepts/containers/EditConcept.jsx (hunk 6 above)
Comment (brucemakallan) [reply src/components/dictionaryConcepts/containers/EditConcept.jsx]: A reference to a concept cannot be edited directly, therefore, it has to first be deleted from the Dictionary, then Recreated. This prop is being called within the method: `updateConceptReference()`
